### PR TITLE
Updates for New NL Interface

### PR DESCRIPTION
**Pull request by ccoffrin**
@odow, can you give me a hand with how to update the tests of the form?

```
    func = constraint.func
    set = constraint.set
    @test isapprox(set.value, 0.64266 / get_base_flow(data); atol = 1e-4)
    @test isa(set, MOI.EqualTo{Float64})
    @test length(func.terms) == 1
```

### CODE REVIEW

**Comment by odow** [comment]
I'd need to take a deeper look. The tests fail because it looks like it's trying to use HiGHS for a nonlinear model???

https://github.com/lanl-ansi/GasModels.jl/blob/5444b86705830823a5cdf508017564990dfa4b51/test/debug.jl#L27-L29

This just looks like it is testing that `func` is `-1.0 * var_ref`?

I don't see why this would break with the changes.

**Comment by odow** [comment]
Just an FYI that I'm only online partially tomorrow, and then I'm away Fri/Sat/Sun NZ time so it may be next week before I get a proper look.

**Comment by ccoffrin** [comment]
No problem on the delay.  From what I can tell the primary issue is this test `@test isa(set, MOI.EqualTo{Float64})` probably the `set` type needs to change due to the new NL data structure.

**Comment by odow** [comment]
This was what I needed to get tests passing: https://github.com/lanl-ansi/Juniper.jl/issues/263
```diff
diff --git a/test/runtests.jl b/test/runtests.jl
index f92f38f..c10df8b 100644
--- a/test/runtests.jl
+++ b/test/runtests.jl
@@ -34,6 +34,8 @@ juniper_solver = JuMP.optimizer_with_attributes(
     "nl_solver" => ipopt_solver,
     "mip_solver" => highs_solver,
     "log_levels" => [],
+    # TODO(odow): bug in Juniper
+    "feasibility_pump" => false,
 )
 
 
```

**Comment by odow** [comment]
This is the patch for the documentation
```diff
diff --git a/docs/Project.toml b/docs/Project.toml
index dfa65cd..3a52a5d 100644
--- a/docs/Project.toml
+++ b/docs/Project.toml
@@ -1,2 +1,5 @@
 [deps]
 Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+
+[compat]
+Documenter = "0.27"
```

**Comment by odow** [comment]
Doc fix is https://github.com/lanl-ansi/GasModels.jl/pull/257

Juniper just needs a new tag: https://github.com/lanl-ansi/Juniper.jl/pull/266

File: Project.toml [hunk 1]
```diff
@@ -11,34 +11,31 @@ InfrastructureModels = "2030c09a-7f63-5d83-885d-db604e0e9cc0"
...
 [compat]
-HiGHS = "~0.3, ~1"
 Dierckx = "~0.4, ~0.5"
+HiGHS = "~0.3, ~1"
 InfrastructureModels = "~0.6, ~0.7"
 Ipopt = "~0.8, ~0.9, ~1"
 JSON = "~0.18, ~0.19, ~0.20, ~0.21"
-JuMP = "~0.22, ~0.23, 1"
+JuMP = "1.15"
 Juniper = ">= 0.4"
```
Comment (odow) [inline Project.toml:27]:
```suggestion
Juniper = "0.9.2"
```
Once https://github.com/lanl-ansi/Juniper.jl/pull/266 is tagged

**Comment by odow** [comment]
@ccoffrin try rebasing this on `master` to get the doc fix, and bump the compat for Juniper

**Pull request merged**